facebook/facebook-ios-sdk
# Adding more specific preprocessor definition for Cocoapod usage

Thanks for proposing a pull request!

To help us review the request, please complete the following:


## Pull Request Details

Earlier naive implementation did not account for a setup where SPM was used for the Facebook iOS SDK and Cocoapods was used for other pods in the same project.

Fixed this by adding a more specific preprocessor macro `FBSDKCOCOAPODS` to the podspecs so that we can gate logic specifically on whether the Facebook iOS SDK is installed via CocoaPods.

## Test Plan

Test Plan: 
1. Create a sample project that includes CocoaPods and install an arbitrary pod that is not one of the iOS SDK Kits.
2. Install the iOS SDK using SPM pointed at this branch.
3. Confirm that it builds
4. Remove the Swift Package and add the iOS SDK Kits as CocoaPods pointed to a local checkout of this branch. ex:
```
  pod 'FBSDKCoreKit', :path => '/Users/<you>/facebook-ios-sdk'
  pod 'FBSDKLoginKit', :path => '/Users/<you>/facebook-ios-sdk'
  pod 'FBSDKShareKit', :path => '/Users/<you>/facebook-ios-sdk'
```
5. Confirm that it builds.
